rust/template/differential_datalog/record.rs

```diff
@@ -1,6 +1,6 @@
 //! An untyped representation of DDlog values and database update commands.
 
-#![allow(unknown_lints)]
+#![allow(unknown_lints,clippy::needless_range_loop)]
```

Comment:
Actually, Rust is correct, you should probably use an iterator, not a range loop.  I would do that rather than disable the warning.